pkg/rpcclient/wsclient.go

```diff
@@ -36,7 +36,10 @@ type WSClient struct {
 	// be closed, so make sure to handle this.
 	Notifications chan Notification
 
-	ws          *websocket.Conn
+	ws     *websocket.Conn
+	ctx    context.Context
+	cancel func()
+
 	done        chan struct{}
 	requests    chan *neorpc.Request
 	shutdown    chan struct{}
```

Comment:
Can `ctx` replace `shutdown` now?

Reply:
I'll check it, can we move this to a separate issue?

Reply:
I've refactored this code, currently WSClient doesn't have its own context. And after we've moved waiting logic out of the client, I'm not sure it's a good way to cancel transaction awaiting. What do you think about
```
Waiter interface {
		Wait(ctx context.Context, h util.Uint256, vub uint32, err error) (*state.AppExecResult, error)
	}
```
?

Reply:
We need `Wait(hash, vub, err)` to use it as a method wrapper. But then waiters can also have `Wait(context, vub, hashes...)` (and it can be exposed through `Actor` too) because in fact we also need `Wait(hash, hash, vub, err)` for the `notary.Actor`.

Reply:
Let's keep shutdown internally for now.